packages/gasket-plugin-elastic-apm/index.js

```diff
@@ -0,0 +1,38 @@
+const { filterSensitiveCookies } = require('./lib/cookies');
+
+/**
+ * Determines if the Elastic APM agent has sufficient config to be active
+ * @param {object} config gasket config
+ * @param {object<string,any>} env environment variables
+ * @returns {boolean} A combined config object
+ */
+const isActive = (config, env) => {
+  const { serverUrl, secretToken } = config || {};
+
+  const combined = {
+    serverUrl: serverUrl || env.ELASTIC_APM_SERVER_URL,
+    secretToken: secretToken || env.ELASTIC_APM_SECRET_TOKEN
+  };
+
+  if (combined.serverUrl && combined.secretToken) {
+    return true;
+  }
+
+  return false;
+};
+
+module.exports = {
+  hooks: {
+    preboot: {
+      handler: async ({ config }) => {
+        require('elastic-apm-node')
+          .start({
+            // eslint-disable-next-line no-process-env
+            active: isActive(config.elasticAPM || {}, process.env),
+            ...config.elasticAPM
```

Comment:
Right now, the env vars are only going to be used in your `isActive` check. I imagine you'll want these vars passed along to the `.start` method, correct? You should probably have a [configure lifecycle hook](https://gasket.dev/#/packages/gasket-plugin-command/README?id=configure) that handles the combining gasket config and possible environment variables.

Something like:

```js
configure(gasket, config) {
  const { elasticAPM = {} } = config;
  const {
    serverUrl = env.ELASTIC_APM_SERVER_URL,
    secretToken = env.ELASTIC_APM_SECRET_TOKEN
  } = elasticAPM;

  config.elasticAPM = {
    serverUrl,
    secretToken,
    ...elasticAPM
  }

  return config;
}
```

Then when it gets to your `preboot`, you can just pass along the resulting `config.elasticAPM`.

Also, consider just doing your active check in the `configure` lifecycle and return it as part of the fixed-up config object (`config.elasticAPM.active`).

Reply:
The Elastic APM Agent already includes lots of logic to reconcile a mix of config options provided to the `.start` method as either literal values or environment variables, and I don't think it would be wise to try to duplicate that effort.

For example, if I were to call `apm.start({})` (with an empty object) but had set the `ELASTIC_APM_SERVER_URL` environment variable, the APM agent would be configured with the server URL from the environment variable without any additional code written on my part. I'd like to preserve that native behavior of the agent so that users can understand how to configure it by referencing the Elastic docs, and not have to learn a separate, "Gasket way".

The `isActive` section tries to inspect both of these config sources to determine the right value for that flag, but other than that leaves Elastic's own config process untouched.

Reply:
You're introducing a Gasket plugin and the ability to configure `elastic-apm-node` with the gasket config. You may as well use the configure lifecycle to reap the full benefit of bringing this to Gasket in the first place.

If the counter-argument is you don't want to get in the way of `elastic-apm-node`'s config reconciliation, then you should just remove the logic determining the `active` config property which is the dynamic odd-ball here. Shouldn't that really be up to the app to configure anyway?

Reply:
I guess I won't block on it this knowing that `elastic-apm-node` is going to be looking at the env vars anyway, from my initial call out. I'd prefer to see configure lifecycle used here for consistency with other gasket plugins. Since apps can override what is dynamically determined for `active` via config then this should be fine.

Reply:
> If the counter-argument is you don't want to get in the way of elastic-apm-node's config reconciliation, then you should just remove the logic determining the active config property which is the dynamic odd-ball here. Shouldn't that really be up to the app to configure anyway?

This is a great point. Managing the `active` property does go against my premise of leaving the agent untouched, but my reason for stepping in there is to improve the developer experience of a common first-run problem...if the agent is started with zero config it will attempt to send APM data to `localhost:8200` which most likely doesn't exist and will cause the agent to start spewing errors. I think this could confuse developers into thinking they've made a mistake in installing the plugin, or that they need a local ES cluster to develop effectively. I think it's a better fallback behavior for the APM agent to be silent and idle in this case.

I've pushed b5368a4 to leverage the configure lifecycle as suggested and allow the user to set the `active` property directly if desired.